From 81ab59876c762cb318938e02186a4bf845754ca7 Mon Sep 17 00:00:00 2001
From: radomir-sebek <rasho.ieee@gmail.com>
Date: Thu, 3 May 2018 21:52:43 +0200
Subject: [PATCH 08/22] [Improvement] Check does DC has enough funds to create
 specific offer (#96)

Check does DC has enough fonds to create specific offer
---
 modules/DCService.js | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/modules/DCService.js b/modules/DCService.js
index 8fe248c..3598ba2 100644
--- a/modules/DCService.js
+++ b/modules/DCService.js
@@ -5,7 +5,7 @@ const Graph = require('./Graph');
 const Blockchain = require('./BlockChainInstance');
 const bytes = require('utf8-length');
 const BN = require('bn.js');
-
+const deasync = require('deasync-promise');
 const Utilities = require('./Utilities');
 const Models = require('../models');
 
@@ -19,6 +19,7 @@ const tenderDuration = biddingTime + 1000;
 const minNumberOfBids = 1;
 const minStakeAmount = new BN('100');
 const maxTokenAmount = new BN('10000000');
+
 /**
  * DC operations (handling new offers, etc.)
  */
@@ -30,8 +31,19 @@ class DCService {
             console.log('Error: ', e);
         });
 
+        let price;
         const importSizeInBytes = new BN(this._calculateImportSize(vertices));
-        const price = `${Utilities.getRandomIntRange(1, 10).toString()}00`;
+        const availableFunds = deasync(Utilities.getAlphaTracTokenBalance());
+        const potentialPrice = `${Utilities.getRandomIntRange(1, 10).toString()}00`;
+
+        if (Number(potentialPrice) > Number(availableFunds)) {
+            // not enought funds in wallet, give all-in for a price
+            price = availableFunds;
+        } else {
+            // sufficient funds, go further
+            price = potentialPrice;
+        }
+
         Models.offers.create({
             id: dataId,
             data_lifespan: totalEscrowTime,
@@ -130,7 +142,7 @@ class DCService {
                                         const eventDcWallet = event.returnValues.DC_wallet;
 
                                         if (Number(eventDataId) === dataId
-                            && eventDcWallet === config.node_wallet) {
+                                            && eventDcWallet === config.node_wallet) {
                                             log.info(`Offer for data ${dataId} successfully finalized`);
                                             DCService.handleFinalizedOffer(dataId);
                                             return true;
-- 
2.13.5 (Apple Git-94)

